src/components/common/StoreReview.tsx

```diff
@@ -0,0 +1,141 @@
+import { useEffect, useState } from 'react';
+import * as ExpoStoreReview from 'expo-store-review';
+import { Trans, t } from '@lingui/macro';
+import { differenceInDays } from 'date-fns';
+
+import { showToast } from './Toaster';
+
+import ImprovementForm from '~components/store-review/ImprovementForm';
+import {
+  BottomSheet,
+  FillButton,
+  OutlineButton,
+  Stack,
+  Text,
+} from '~components/uikit';
+import storage from '~utils/storage';
+import { styled } from '~styles';
+
+const DAYS_TO_WAIT_BEFORE_ASKING_AGAIN = 7; // 7 days
+const DAYS_BETWEEN_REVIEWS = 365; // 1 year
+
+export default function StoreReview() {
+  const [open, setOpen] = useState<boolean>(false);
+  const [improvementRequest, setImprovementRequest] = useState<boolean>(false);
+
+  useEffect(() => {
+    const persistedHasReviewed = storage.getNumber('@app/last-review-done');
+    const getReviewDate = async () => {
+      if (await ExpoStoreReview.hasAction()) {
+        const persistedLastAsked = storage.getNumber(
+          '@app/last-requested-review'
+        );
+        if (persistedLastAsked) {
+          const differenceInDaysValue = differenceInDays(
+            new Date(),
+            new Date(persistedLastAsked)
+          );
+          // Check if the difference is a week or more
+          const isWeekOrMoreOld =
+            differenceInDaysValue >= DAYS_TO_WAIT_BEFORE_ASKING_AGAIN;
+
+          if (!isWeekOrMoreOld) {
+            setOpen(true);
+            updateLastAsked();
+          }
+        } else {
+          updateLastAsked();
+        }
+      }
+    };
+
+    // Check if the user has already reviewed the app in the last year
+    if (
+      persistedHasReviewed &&
+      differenceInDays(new Date(), new Date(persistedHasReviewed)) >=
+        DAYS_BETWEEN_REVIEWS
+    ) {
+      setTimeout(() => {
+        getReviewDate();
+      }, 10); // Wait for 10 seconds before asking for review (To be removed once we implement it in a strategic place)
```

Comment:
This is not 10 seconds but 10 milliseconds 😄

Reply:
Oups, forgot to put it back after my tests